lib/amqp.js

```diff
@@ -125,26 +125,36 @@ class Amqp {
 
     async publishMessage(exchangeName, routingKey, payloadBuffer, options, iteration) {
         const settings = this.settings;
-        const result = this.publishChannel.publish(exchangeName, routingKey, payloadBuffer, options);
-        if (!result) {
-            log.warn('Buffer full when publishing a message to '
-                + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
+        const publishChannel = this.publishChannel;
+
+        function promisifiedPublish(exchangeName, routingKey, payloadBuffer, options) {
+            let result;
+            return new Promise((resolve, reject) => {
+                result = publishChannel.publish(exchangeName, routingKey, payloadBuffer, options, (err, ok) => {
+                    err ? reject(err) : resolve(ok);
+                });
+            })
+                .then(() => result);
         }
 
         try {
-            await this.publishChannel.waitForConfirms();
+            const result = await promisifiedPublish(exchangeName, routingKey, payloadBuffer, options);
+            if (!result) {
+                log.warn('Buffer full when publishing a message to '
+                    + 'exchange=%s with routingKey=%s', exchangeName, routingKey);
+            }
+            return result;
+
         } catch (error) {
             log.error('Failed on publishing message to queue');
```

Comment:
It's not a part of this task, but it will be nice to log an error too.
```suggestion
            log.error(error, 'Failed on publishing message to queue');
```